crates/aiken-lang/src/gen_uplc.rs

```diff
@@ -4537,30 +4499,27 @@ impl<'a> CodeGenerator<'a> {
                     | DefaultFunction::Trace
                     | DefaultFunction::ChooseList
                     | DefaultFunction::ChooseData
+                    | DefaultFunction::MkCons
                     | DefaultFunction::UnConstrData => {
-                        builder::special_case_builtin(&func, count, arg_vec)
+                        builder::special_case_builtin(&func, tipo, count, arg_vec)
                     }
-
                     DefaultFunction::FstPair | DefaultFunction::SndPair => {
                         builder::undata_builtin(&func, count, ret_tipo, arg_vec)
                     }
-
                     DefaultFunction::HeadList if !tipo.is_pair() => {
                         builder::undata_builtin(&func, count, ret_tipo, arg_vec)
                     }
-
-                    DefaultFunction::MkCons | DefaultFunction::MkPairData => {
-                        unimplemented!(
-                            "MkCons and MkPairData should be handled by an anon function or using [] or ( a, b, .., z).\n"
-                        )
-                    }
                     _ => {
                         let mut term: Term<Name> = func.into();
 
                         term = builder::apply_builtin_forces(term, func.force_count());
 
-                        for arg in arg_vec {
-                            term = term.apply(arg.clone());
+                        if func.arg_is_unit() {
```

Comment:
Note: I had to resort to this new ad-hoc method because ...

1. We cannot trust the type `tipo` since it sometimes contain the full type, sometimes just the return type. I don't know how we ended here abusing this type instead of introducing another parameter but that semantic confusion makes it quite impractical. 

2. We cannot rely on the `args_vec` vector either because in the case where the function is simply return as is, that same op code is summoned even if the function hasn't been invoked (and so, it happens many times that functions are passed without arguments and that the arg_vec is empty, despite the function having actual arguments). 

For lack of better solution, this seems to work fine.